tiledb/api/src/query/subarray.rs

```diff
@@ -63,45 +184,134 @@ where
+                                    start.as_ptr() as *const std::ffi::c_void,
+                                    end.as_ptr() as *const std::ffi::c_void,
+                                    std::ptr::null(),
+                                )
+                            })?;
+                        }
+                    }
                 })
             }
-            LookupKey::Name(name) => {
-                let c_name = cstring!(name);
-                self.capi_return(unsafe {
-                    ffi::tiledb_subarray_add_range_by_name(
-                        c_context,
-                        c_subarray,
-                        c_name.as_ptr(),
-                        c_start,
-                        c_end,
-                        std::ptr::null(),
-                    )
+            Range::Multi(_) => unreachable!(
+                "This is rejected by range.check_dimension_compatibility"
+            ),
+            Range::Var(range) => {
+                var_value_range_go!(range, _DT, start, end, {
+                    match key.into() {
+                        LookupKey::Index(idx) => {
+                            ctx.capi_return(unsafe {
+                                ffi::tiledb_subarray_add_range_var(
+                                    c_ctx,
+                                    c_subarray,
+                                    idx as u32,
+                                    start.as_ptr() as *const std::ffi::c_void,
+                                    start.len() as u64,
+                                    end.as_ptr() as *const std::ffi::c_void,
+                                    end.len() as u64,
+                                )
+                            })?;
+                        }
+                        LookupKey::Name(name) => {
+                            let c_name = cstring!(name);
+                            ctx.capi_return(unsafe {
+                                ffi::tiledb_subarray_add_range_var_by_name(
+                                    c_ctx,
+                                    c_subarray,
+                                    c_name.as_ptr(),
+                                    start.as_ptr() as *const std::ffi::c_void,
+                                    start.len() as u64,
+                                    end.as_ptr() as *const std::ffi::c_void,
+                                    end.len() as u64,
+                                )
+                            })?;
+                        }
+                    }
                 })
             }
-        }?;
+        }
+
+        Ok(self)
+    }
+
+    /// Add a list of point ranges to the query.
+    pub fn add_point_ranges<Key: Into<LookupKey>, T: CAPISameRepr>(
+        self,
+        key: Key,
+        points: &[T],
+    ) -> TileDBResult<Self> {
+        let schema = self.query.base().query.array.schema()?;
+        let dim_idx = schema.domain()?.dimension_index(key)?;
+
```

Comment:
Some kind of datatype compatibility check oughta go here

Reply:
Good call. And easy enough.

Reply:
[Check the datatype for point ranges.](https://github.com/TileDB-Inc/tiledb-rs/pull/72/commits/dbd194cf69f6a423cd84f8eb65efc77381a91eb0)